### PR TITLE
Kill yourself nerds...

### PR DESCRIPTION
**Pull request by ghost**
... Physically. No just typing Suicide anymore.

On Sage atleast.

🆑 BeloneX
config: Removes the suicide verb on sage. You'll need to actually just kill yourself, or ghost.
tweaked: Shooting yourself in the mouth now deals 10* the damage which is guranteed to kill you in almost all scenarios.
/🆑

Disclaimer: Don't actually kill yourself.

### CODE REVIEW

**Comment by ToasterBan** [comment]
typing suicide should just teleport you to cryo

**Comment by Penwin0** [comment]
I hate you

**Comment by r1ks-iwnl** [comment]
How am i supposed to rocket jump to death now using the PML

**Comment by VictorVow** [comment]
I'd rather see it be time-locked for the first 20 mins of the shift than removed completely. There's not any RP-friendly reason why someone would need to do it shiftstart but I can see it being reasonable later after certain events happen.

**Comment by ghost** [comment]
> I'd rather see it be time-locked for the first 20 mins of the shift than removed completely. There's not any RP-friendly reason why someone would need to do it shiftstart but I can see it being reasonable later after certain events happen.

Literally what situations? In whatever situation you're imagining, where suicide would be reasonable, they would have access to a less RP damaging method than typing in a verb and magically falling over dead.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
make it a config flag to disable completely.

**Comment by ghost** [comment]
> make it a config flag to disable completely.

It is though?

**Comment by Sinestia** [comment]
> > I'd rather see it be time-locked for the first 20 mins of the shift than removed completely. There's not any RP-friendly reason why someone would need to do it shiftstart but I can see it being reasonable later after certain events happen.
> 
> Literally what situations? In whatever situation you're imagining, where suicide would be reasonable, they would have access to a less RP damaging method than typing in a verb and magically falling over dead.

all of that flavour text tho :( :(

**Comment by Sinestia** [comment]
> I'd rather see it be time-locked for the first 20 mins of the shift than removed completely. There's not any RP-friendly reason why someone would need to do it shiftstart but I can see it being reasonable later after certain events happen.

It already seems to do that apparently, but for the first 15 minutes

**Comment by PowerfulBacon** [comment]
Also does suicide free up job slots? If so then suicide is better than people just leaving on MRP, since if they aren't going to go to cryo, suiciding is better than just leaving.

**Comment by ghost** [comment]
> Also does suicide free up job slots? If so then suicide is better than people just leaving on MRP, since if they aren't going to go to cryo, suiciding is better than just leaving.

No, it doesn't.

**Comment by ghost** [comment]
> make it a config flag to disable completely.

Done.

File: code/modules/client/verbs/suicide.dm [hunk 1]
```diff
@@ -277,5 +277,7 @@
 			if(src.job)
 				message_admins("[key_name(src)] (job: [src.job]) attempted to commit suicide at [AREACOORD(src)]. Time until suicide is possible: [DisplayTimeText(timeleft)].")
 			return
+	if(CONFIG_GET(flag/disabled_suicide))
+		(alert("No, enter the cryopods or contact the admins if you have a valid reason."))
```
Comment (PowerfulBacon) [inline code/modules/client/verbs/suicide.dm]:
```suggestion
		alert("No, enter the cryopods or contact the admins if you have a valid reason.")
```
Also change the message itself to something like "<span class='warning'>Suiciding is disabled, please exit the game via the cryopods or contact the admins.</span>"
Change the mess

**Comment by eeSPee** [comment]
What is this?

I see people having... FUN?!

Oh no we must speedmerge something imediately.

EDIT I might just do it if this PR makes it through

**Comment by Archanial** [comment]
Cringe pr
Suicide is badass

**Comment by ghost** [comment]
> Cringe pr
> Suicide is badass

Do you have anything of sustance to say or are you only here for off topic comments?

**Comment by Archanial** [comment]
> > Cringe pr
> > Suicide is badass
> 
> Do you have anything of sustance to say or are you only here for off topic comments?

You are fixing something that is not an issue and I can easily imagine situations where you'd want to suicide. 

This only exists because of Bloons banning 2 people. 

This further separates golden from sage which is bad for playerbase.

**Comment by ghost** [comment]
> You are fixing something that is not an issue and I can easily imagine situations where you'd want to suicide.

I already debunked why that is a bad argument above.
> This only exists because of Bloons banning 2 people.

No, I've had my eyes on this for a long time.
> This further separates golden from sage which is bad for playerbase.

Fair enough, but you forget that we are running a MRP server here.

**Comment by Sinestia** [comment]
even bay hasn't done this smh hh h

**Comment by ghost** [comment]
all of those flavour texts..... all cause muh rp. unbased belone

**Comment by Archanial** [comment]
> RP
> OOC flavour texts

???

**Comment by TheFakeElon** [comment]
suicide is already a config option and it's disabled on sage lol

**Comment by ghost** [comment]
just make it so if you suicide the job you had shift start frees up smh

**Comment by ItsTisme** [comment]
> just make it so if you suicide the job you had shift start frees up smh

This, its not much harder to type *suicide then it is to walk out the arrivals airlock. This wont changre anyhting

**Comment by PowerfulBacon** [comment]
I thought I made a comment on this but I guess it didn't go through.
Assuming suicide frees up a job slot, then suiciding is a lot better than players killing themselves by hitting themselves with a toolbox or walking out and airlock and better than them just ghosting since their job will be freed. However, it is likely a possibility that suicide doesn't free up job slots (in which case it should.). Players will always choose the easiest way to leave the game if they can, and will just close the game leaving their player standing mindlessly in the halls, wasting a job slot up rather than crying

Edit: nevermind I just saw my other comment

File: code/modules/client/verbs/suicide.dm (hunk 1 above)
Comment (ghost) [reply code/modules/client/verbs/suicide.dm]: Done.

**Comment by knacker48** [comment]
> In whatever situation you're imagining, where suicide would be reasonable, they would have access to a less RP damaging method than typing in a verb and magically falling over dead.

Dude this doesn't stop people from committing suicide, It just draws it out. Instead of just slitting their throat by having a knife and typing suicide they'll just start applying a toolbox to the head 15 times. Do you really think that this will improves RP in any meaningful way?

**Comment by ItsTisme** [comment]
Would be nice if SSD mobs automaticly walked to the cryo pods or something to free up job slots or a "cryo" button

**Comment by PowerfulBacon** [comment]
I think its probably best to let the headmins have the say over this one

**Comment by Sinestia** [comment]
> Would be nice if SSD mobs automaticly walked to the cryo pods or something to free up job slots or a "cryo" button

Para has a system that snatches you with a teleporter after like, 20

that could be ported eventually

**Comment by OceanFish1** [comment]
Cringe pr

Cringe pr

**Comment by nikk735** [comment]
wastes all the work people put into custom suicides

**Comment by qwertyquerty** [comment]
Not a solution to the problem

**Comment by ghost** [comment]
> Not a solution to the problem

Did you even consolidate with the Headmins before closing this? They were supposed to decide on it.

**Pull request closed without merging**